trackintel/preprocessing/triplegs.py

```diff
@@ -205,6 +209,21 @@ def _seperate_ids(row):
     # delete activities
     trips = trips_with_act[~trips_with_act["activity"]].copy()
 
+    # fill NaN start and end points with start of first tripleg (or end of last tripleg) on the trip
+    def geom_by_tripleg(trips, tpls, is_origin):
+        point_idx = 0 if is_origin else -1
+        col = "origin_staypoint_id" if is_origin else "destination_staypoint_id"
+        nan_geom_cols = trips[pd.isna(trips[col])].copy()
+        nan_geom_cols.loc[:, "tpls_id"] = nan_geom_cols.tpls.map(lambda x: x[point_idx])
+        nan_geom_cols = nan_geom_cols.join(tpls, on="tpls_id", how="left", lsuffix="_l", rsuffix="_r")
+        nan_geom_cols["geom_point"] = nan_geom_cols.geom_r.apply(lambda x: x.boundary[point_idx])
+        return nan_geom_cols["geom_point"]
```

Comment:
I did use the triplegs id now to join the two tables, in order to fill in the points where the staypoint is NaN. Maybe there is a smarter way to do this.